tests/integration/states/file.py

```diff
@@ -290,9 +340,16 @@ def test_managed_file_with_grains_data(self):
         Test to ensure we can render grains data into a managed
         file.
         '''
-        ret = self.run_function('state.sls', ['file-grainget'])
-        self.assertTrue(os.path.exists('/tmp/file-grain-test'))
-        file_contents = open('/tmp/file-grain-test', 'r').readlines()
+        if IS_WINDOWS:
+            grain_path = 'C:\\Windows\\Temp\\file-grain-test'
+        else:
+            grain_path = '/tmp/file-grain-test'
+
+        state_file = 'file-grainget'
+
+        ret = self.run_function('state.sls', [state_file])
+        self.assertTrue(os.path.exists(grain_path))
+        file_contents = open(grain_path, 'r').readlines()
```

Comment:
Since you are not using a context manager here, you need to explicitly close this file handle when you are done with it.